### PR TITLE
Sign in - mobile header

### PR DESCRIPTION
**Pull request by zeftilldeath**
Bringing my account and sign in to the header at mobile breakpoints.

- This means sign in and my account have been removed from the veggie burger menu.
- When you open the dropdown dialogue page scroll is prevented until you click the trigger or outside of the dropdown.
- Dropdown style has been kept as consistent with desktop state as possible but with refinements for mobile, such as full width, blackout behind dropdown and indication arrow.
- Dropdown scrollable if device is rotated.

# Larger devices
frame 4

# Smaller devices
frame 5

### CODE REVIEW

**Comment by zeftilldeath** [comment]
@SiAdcock If there are any accessibility points we can check for at the mobile breakpoints please let me know

File: common/app/views/fragments/header.scala.html [hunk 1]
```diff
@@ -75,12 +75,11 @@
                         }
 
                         @if(editionId == "uk") {
-                            <a class="top-bar__item hide-until-tablet"
+                            <a class="top-bar__item hide-until-desktop"
```
Comment (NataliaLKB) [inline common/app/views/fragments/header.scala.html]: What is the reason for this change / has it been checked with the relevant people?

File: common/app/views/fragments/nav/headerMenu.scala.html [hunk 2]
```diff
@@ -120,8 +120,6 @@
                         </a>
                     </li>
                 }
-
-                @fragments.nav.userAccountLinks()
```
Comment (NataliaLKB) [inline common/app/views/fragments/nav/headerMenu.scala.html]: Nice! Can we also delete this file? I don't think it is being used by anything else

File: common/app/views/fragments/header.scala.html (hunk 1 above)
Comment (zeftilldeath) [reply common/app/views/fragments/header.scala.html]: The change doesn't affect what users see. The word dating was hidden until desktop but oddly it's parent was hidden until tablet, so I just removed the nested hides, to prevent an empty parent sitting there.

File: static/src/javascripts/projects/common/modules/navigation/new-header.js [hunk 3]
```diff
@@ -233,6 +233,8 @@ const toggleMenu = (): void => {
 };
 
 const toggleDropdown = (menuAndTriggerEls: MenuAndTriggerEls): void => {
+    const documentElement = document.documentElement;
+    const globalOpenClass = 'dropdown-is-open';
```
Comment (NataliaLKB) [inline static/src/javascripts/projects/common/modules/navigation/new-header.js]: dropdown--open would be more BEM

File: static/src/stylesheets/layout/nav/_my-account.scss [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    svg {
+        height: 18px;
+        width: 18px;
+
+        @include mq(mobileMedium) {
+            height: 23px;
+            width: 23px;
+        }
+
+        path {
```
Comment (NataliaLKB) [inline static/src/stylesheets/layout/nav/_my-account.scss]: I think if you remove the colour from the path in the svg, you then don't have to nest the `fill` in path

**Comment by jfsoul** [comment]
superb work @zeftilldeath - thank you! 💎 

Do you have a screenshot for the signed-out state?

I will let someone more knowledgable approve, but it looks great to me.

**Comment by zeftilldeath** [comment]
Thanks @jfsoul Same icon and appearance for signed in and signed out at the moment due to lack of space

**Comment by jfsoul** [comment]
> Same icon and appearance for signed in and signed out at the moment due to lack of space

Gotcha, that makes sense.  So clicking the icon when signed out opens the menu with just a link to sign-in/register?

File: static/src/stylesheets/layout/nav/_top-bar-dropdown.scss [hunk 5]
```diff
@@ -32,7 +50,35 @@
...
+
+    @include mq($until: tablet) {
+        position: fixed;
+        border-radius: 0;
+        top: 36px;
+        left: 0;
+        right: 0;
+        width: auto;
+
+        @media (max-height: 370px) {
```
Comment (NataliaLKB) [inline static/src/stylesheets/layout/nav/_top-bar-dropdown.scss]: A comment here would be nice explaining the media query

File: static/src/stylesheets/layout/nav/_top-bar-dropdown.scss [hunk 6]
```diff
@@ -42,17 +88,22 @@
 
 .dropdown-menu__title {
     @include fs-textSans(4);
+    font-size: 16px;
```
Comment (NataliaLKB) [inline static/src/stylesheets/layout/nav/_top-bar-dropdown.scss]: If she are going to overwrite the font-size let's not include the mixin above

**Review by NataliaLKB — APPROVED** [review]

File: static/src/stylesheets/layout/nav/_top-bar-dropdown.scss (hunk 5 above)
Comment (walaura) [reply static/src/stylesheets/layout/nav/_top-bar-dropdown.scss]: Is the mq necessary? Would `min-height: calc(100%-36px)` (maybe `min-height:90%` for older browsers too) + always overflowing to auto) work? That way we don't have `370px` set in stone and if the menu grows that's okay
